### PR TITLE
Allow our Monolog implementation to be 1.x and 2.x compatible | TEC-3638

### PR DESCRIPTION
**Pull request by borkweb**
### Core issue

Our Monolog implementation assumes Monolog 1.x. WooCommerce Bookings (the plugin whose conflict was flagged most recently) includes Monolog 2.x. Our implementation is not compatible with 2.x due to a couple of reasons.

#### Improper instantiation

Our instantiation of the `Monolog\Handler\ErrorHandler` is incorrect as per the docblock in both versions 1.x and 2.x of Monolog. The first argument should be an `int` and we were passing `null`. With Monolog 2.x, the type hinting enforces the use of an `int` where we should have been using one all along.

#### Our `Canonical_Formatter`

We appropriately extend the `Monolog\Formatter\LineFormatter` class for our `Canonical_Formatter` class—which is used when instantiating the `Monolog\Handler\ErrorLogHandler` object our `Log\Service_Provider`. However, extending the class as we do makes our `Canonical_Formatter` incompatible with 2.x.

We cannot update to Monolog 2.x (whose minimum PHP version is 7.2) and the Monolog 1.x & 2.x method signatures for the `LineFormatter` class are incompatible.

The fix that I have put in place allows us to skirt around the issue of method signatures between Monolog versions by changing `Canonical_Formatter` to instantiate `LineFormatter` rather than extending it.

### Artifacts

**The Monolog-related tests are passing and that should serve as our validation for the Monolog-related logging behaviors.**

For the display of the expected functionality as per [TEC-3638], here's a screenshot of the rendered event (rather than a WP Fatal):

Selection_066

See: [TEC-3638]
Related: #1452

[TEC-3638]: https://moderntribe.atlassian.net/browse/TEC-3638

### CODE REVIEW

**Review by sc0ttkclark — APPROVED** [review]

**Comment by borkweb** [comment]
@sc0ttkclark, can you give this another look-see?

**Review by sc0ttkclark — APPROVED** [review]
Hey you're good at this or somethiny

**Review by bordoni — APPROVED** [review]

**Comment by bordoni** [comment]
Damn, that is a good find. I did not think of this as possible, I looked at the two diff implementations and assumed. Thanks for finding this.

File: src/Tribe/Log/Canonical_Formatter.php [hunk 1]
```diff
@@ -11,9 +11,37 @@
 namespace Tribe\Log;
 
 
+use Monolog\Formatter\FormatterInterface;
 use Monolog\Formatter\LineFormatter;
 
-class Canonical_Formatter extends LineFormatter {
+class Canonical_Formatter implements FormatterInterface {
```
Comment (lucatume) [inline src/Tribe/Log/Canonical_Formatter.php:17]: Composition over inheritance, one can rarely go wrong. Thanks for this fix.

**Review by lucatume — APPROVED** [review]

**Pull request merged**